Sofa/framework/DefaultType/src/sofa/defaulttype/RigidCoord.h

```diff
@@ -267,15 +267,17 @@ class RigidCoord<3,real>
         orientation.fromMatrix(rot);
     }
 
-    /// Write to the given matrix
-    template<class Mat>
-    void toMatrix(Mat& m) const
+    /// Write to the given 3x3 matrix
+    void toMatrix( sofa::type::Mat<3,3,real>& m) const
     {
         m.identity();
         orientation.toMatrix(m);
-        m[0][3] = (typename Mat::Real)center[0];
-        m[1][3] = (typename Mat::Real)center[1];
-        m[2][3] = (typename Mat::Real)center[2];
+    }
+
+    /// Write to the given 4x4 matrix
+    void toMatrix( sofa::type::Mat<4,4,real>& m) const
+    {
+        toHomogeneousMatrix(m);
```

Comment:
It seems that the translation part has been forgotten no ?

Reply:
The `toHomogeneousMatrix` function is just bellow. Do you see something wrong?

Reply:
In the two versions, the translation part is not set